scapy/contrib/iec104_information_elements.py

```diff
@@ -0,0 +1,3059 @@
+    QRP_FLAG_COMPATIBLE_RESERVED_25 = 25
+    QRP_FLAG_COMPATIBLE_RESERVED_26 = 26
+    QRP_FLAG_COMPATIBLE_RESERVED_27 = 27
+    QRP_FLAG_COMPATIBLE_RESERVED_28 = 28
+    QRP_FLAG_COMPATIBLE_RESERVED_29 = 29
+    QRP_FLAG_COMPATIBLE_RESERVED_30 = 30
+    QRP_FLAG_COMPATIBLE_RESERVED_31 = 31
+    QRP_FLAG_COMPATIBLE_RESERVED_32 = 32
+    QRP_FLAG_COMPATIBLE_RESERVED_33 = 33
+    QRP_FLAG_COMPATIBLE_RESERVED_34 = 34
+    QRP_FLAG_COMPATIBLE_RESERVED_35 = 35
+    QRP_FLAG_COMPATIBLE_RESERVED_36 = 36
+    QRP_FLAG_COMPATIBLE_RESERVED_37 = 37
+    QRP_FLAG_COMPATIBLE_RESERVED_38 = 38
+    QRP_FLAG_COMPATIBLE_RESERVED_39 = 39
+    QRP_FLAG_COMPATIBLE_RESERVED_40 = 40
+    QRP_FLAG_COMPATIBLE_RESERVED_41 = 41
+    QRP_FLAG_COMPATIBLE_RESERVED_42 = 42
+    QRP_FLAG_COMPATIBLE_RESERVED_43 = 43
+    QRP_FLAG_COMPATIBLE_RESERVED_44 = 44
+    QRP_FLAG_COMPATIBLE_RESERVED_45 = 45
+    QRP_FLAG_COMPATIBLE_RESERVED_46 = 46
+    QRP_FLAG_COMPATIBLE_RESERVED_47 = 47
+    QRP_FLAG_COMPATIBLE_RESERVED_48 = 48
+    QRP_FLAG_COMPATIBLE_RESERVED_49 = 49
+    QRP_FLAG_COMPATIBLE_RESERVED_50 = 50
+    QRP_FLAG_COMPATIBLE_RESERVED_51 = 51
+    QRP_FLAG_COMPATIBLE_RESERVED_52 = 52
+    QRP_FLAG_COMPATIBLE_RESERVED_53 = 53
+    QRP_FLAG_COMPATIBLE_RESERVED_54 = 54
+    QRP_FLAG_COMPATIBLE_RESERVED_55 = 55
+    QRP_FLAG_COMPATIBLE_RESERVED_56 = 56
+    QRP_FLAG_COMPATIBLE_RESERVED_57 = 57
+    QRP_FLAG_COMPATIBLE_RESERVED_58 = 58
+    QRP_FLAG_COMPATIBLE_RESERVED_59 = 59
+    QRP_FLAG_COMPATIBLE_RESERVED_60 = 60
+    QRP_FLAG_COMPATIBLE_RESERVED_61 = 61
+    QRP_FLAG_COMPATIBLE_RESERVED_62 = 62
+    QRP_FLAG_COMPATIBLE_RESERVED_63 = 63
+    QRP_FLAG_COMPATIBLE_RESERVED_64 = 64
+    QRP_FLAG_COMPATIBLE_RESERVED_65 = 65
+    QRP_FLAG_COMPATIBLE_RESERVED_66 = 66
+    QRP_FLAG_COMPATIBLE_RESERVED_67 = 67
+    QRP_FLAG_COMPATIBLE_RESERVED_68 = 68
+    QRP_FLAG_COMPATIBLE_RESERVED_69 = 69
+    QRP_FLAG_COMPATIBLE_RESERVED_70 = 70
+    QRP_FLAG_COMPATIBLE_RESERVED_71 = 71
+    QRP_FLAG_COMPATIBLE_RESERVED_72 = 72
+    QRP_FLAG_COMPATIBLE_RESERVED_73 = 73
+    QRP_FLAG_COMPATIBLE_RESERVED_74 = 74
+    QRP_FLAG_COMPATIBLE_RESERVED_75 = 75
+    QRP_FLAG_COMPATIBLE_RESERVED_76 = 76
+    QRP_FLAG_COMPATIBLE_RESERVED_77 = 77
+    QRP_FLAG_COMPATIBLE_RESERVED_78 = 78
+    QRP_FLAG_COMPATIBLE_RESERVED_79 = 79
+    QRP_FLAG_COMPATIBLE_RESERVED_80 = 80
+    QRP_FLAG_COMPATIBLE_RESERVED_81 = 81
+    QRP_FLAG_COMPATIBLE_RESERVED_82 = 82
+    QRP_FLAG_COMPATIBLE_RESERVED_83 = 83
+    QRP_FLAG_COMPATIBLE_RESERVED_84 = 84
+    QRP_FLAG_COMPATIBLE_RESERVED_85 = 85
+    QRP_FLAG_COMPATIBLE_RESERVED_86 = 86
+    QRP_FLAG_COMPATIBLE_RESERVED_87 = 87
+    QRP_FLAG_COMPATIBLE_RESERVED_88 = 88
+    QRP_FLAG_COMPATIBLE_RESERVED_89 = 89
+    QRP_FLAG_COMPATIBLE_RESERVED_90 = 90
+    QRP_FLAG_COMPATIBLE_RESERVED_91 = 91
+    QRP_FLAG_COMPATIBLE_RESERVED_92 = 92
+    QRP_FLAG_COMPATIBLE_RESERVED_93 = 93
+    QRP_FLAG_COMPATIBLE_RESERVED_94 = 94
+    QRP_FLAG_COMPATIBLE_RESERVED_95 = 95
+    QRP_FLAG_
```

Comment:
For all of those, don't you think you could emulate their definition using `__getattr__` ?
Something like (adapt it)
```
def __getattr__(self, attr):
    if attr.startswith("LSQ_FLAG_COMPATIBLE_RESERVED_"):
        return int(attr[29:])
   [...]
```

That would fix the issues of list over-defined

Reply:
seems teaching an old guy new stuff is one of your strengths. keep pushing!

Reply:
addressed in bd9a2d0. all range-based attributes and dict entries are now generated.

I'm using a class decorator for adding the attributes and dict entries instead of __getattr__()  as the latter one would force the user to use an instance of the class. Any drawbacks doing it this way?

Reply:
No that’s also good